File: DesignPatternsExercises/java/IteratorDesignPattern/Exercise/IteratorDesignPattern.java
import java.util.Iterator;

public class IteratorDesignPattern {
	public static void main(String[] args){
		Iterator iterator;
		System.out.println("Set of Countries:");
		iterator = new CountrySet().getIterator();
		displayCountries(iterator);
		System.out.println();

		System.out.println("Array of Countries:");
		iterator = new CountryArray().getIterator();
		displayCountries(iterator);
		System.out.println();

		System.out.println("List of Countries:");
		iterator = new CountryList().getIterator();
		displayCountries(iterator);
	}

	public static void __________(1)__________(Iterator iterator) {
		while (iterator.hasNext()) {
			Country country = __________(2)__________ iterator.next();
			System.out.println("Country: "+ country.getName() + " Area: " + country.getArea() +
					" Population: " +country.getPopulation());
		}
	}
}

File: DesignPatternsExercises/ruby/iterator_design_pattern/solution/iterator_design_pattern.rb
require_relative 'country_array'
require_relative 'country_set'

# def __________(1)__________
def display_countries
  # __________(2)__________.each { |country| puts "Country: #{country.name} -- Area: #{country.area} -- Population: #{country.population}" }
  @iterator.each { |country| puts "Country: #{country.name} -- Area: #{country.area} -- Population: #{country.population}" }
end

puts 'Array of Countries:'
# @iterator = CountryArray.new.__________(3)__________
@iterator = CountryArray.new.country_array
display_countries
puts ''

puts 'Set of Countries:'
# @iterator = CountrySet.new.__________(4)__________
@iterator = CountrySet.new.country_set
display_countries
